File: java/src/main/java/com/antybrowser/sdk/AntyBrowserException.java
package com.antybrowser.sdk;

public class AntyBrowserException extends RuntimeException {
    private final int statusCode;
    private final String responseBody;

    public AntyBrowserException(String message) {
        this(message, 0, null);
    }

    public AntyBrowserException(String message, int statusCode, String responseBody) {
        super(message);
        this.statusCode = statusCode;
        this.responseBody = responseBody;
    }

    public int getStatusCode() { return statusCode; }
    public String getResponseBody() { return responseBody; }
}

File: nuget/Antybrowser.SDK/Exceptions/AntybrowserException.cs
using System;
using System.Net;

namespace Antybrowser.SDK.Exceptions
{
    public class AntybrowserException : Exception
    {
        public HttpStatusCode? StatusCode { get; }
        public string? ResponseContent { get; }

        public AntybrowserException(string message) : base(message)
        {
        }

        public AntybrowserException(string message, Exception innerException) : base(message, innerException)
        {
        }

        public AntybrowserException(string message, HttpStatusCode statusCode, string? responseContent) 
            : base($"{message} (Status: {statusCode})")
        {
            StatusCode = statusCode;
            ResponseContent = responseContent;
        }
    }
}
